src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1087,6 +1108,63 @@ emit_vector_insert_element (
 	return ins;
 }
 
+#if defined(TARGET_ARM64)
+static MonoInst*
+emit_normalize_vector_2_3_4 (MonoCompile *cfg, MonoClass *klass, MonoInst *arg){
+	MonoInst *vec_squared = emit_simd_ins (cfg, klass, OP_XBINOP, arg->dreg, arg->dreg);
+	vec_squared->inst_c0 = OP_FMUL;
+	vec_squared->inst_c1 = MONO_TYPE_R4;
+
+	const char *class_name = m_class_get_name (klass);
+	if (!strcmp ("Plane", class_name)) {
```

Comment:
Should Vector3 be added to this check as well?

Reply:
Vector3 should have 0 on the fourth lane. Plane is special case as it has four valid values, but we need to use its normal vector for calculations.